src/test/java/com/yourcompany/utils/SauceUtils.java

```diff
@@ -25,60 +29,56 @@ public static void UpdateResults(String username, String accessKey, boolean test
+        Map<String, Object> platform = (Map<String, Object>) platforms.get(value);
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        for (String key : platform.keySet()) {
+            capabilities.setCapability(key, platform.get(key));
+        }
+        return capabilities;
+    }
+
     /**
      * Populates the <code>updates</code> map with the value of the system property/environment variable
      * with the following name:
      * <ol>
-     *     <li>SAUCE_BAMBOO_BUILDNUMBER</li>
-     *     <li>JENKINS_BUILD_NUMBER</li>
-     *     <li>BUILD_TAG</li>
-     *     <li>BUILD_NUMBER</li>
-     *     <li>TRAVIS_BUILD_NUMBER</li>
-     *     <li>CIRCLE_BUILD_NUM</li>
+     * <li>SAUCE_BAMBOO_BUILDNUMBER</li>
+     * <li>JENKINS_BUILD_NUMBER</li>
+     * <li>BUILD_TAG</li>
+     * <li>BUILD_NUMBER</li>
+     * <li>TRAVIS_BUILD_NUMBER</li>
+     * <li>CIRCLE_BUILD_NUM</li>
      * </ol>
-     * @param updates String,Object pair containing job updates
      */
-    private static void addBuildNumberToUpdate(Map<String, Object> updates) {
-        //try Bamboo
-        String buildNumber = readPropertyOrEnv("SAUCE_BAMBOO_BUILDNUMBER", null);
-        if (buildNumber == null || buildNumber.equals("")) {
-            //try Jenkins
-            buildNumber = readPropertyOrEnv("JENKINS_BUILD_NUMBER", null);
-        }
-
-        if (buildNumber == null || buildNumber.equals("")) {
-            //try BUILD_TAG
-            buildNumber = readPropertyOrEnv("BUILD_TAG", null);
-        }
+    public static String getBuildName() {
+        List<String> variables = Arrays.asList("SAUCE_BAMBOO_BUILDNUMBER",
+                "JENKINS_BUILD_NUMBER",
+                "CIRCLE_BUILD_NUM",
+                "BUILD_TAG",
+                "BUILD_NUMBER",
+                "TRAVIS_BUILD_NUMBER");
 
-        if (buildNumber == null || buildNumber.equals("")) {
-            //try BUILD_NUMBER
-            buildNumber = readPropertyOrEnv("BUILD_NUMBER", null);
+        for (String variable : variables) {
+            String buildName = readPropertyOrEnv(variable);
+            if (buildName != null && !buildName.equals("")) {
+                return buildName;
+            }
         }
-        if (buildNumber == null || buildNumber.equals("")) {
-            //try TRAVIS_BUILD_NUMBER
-            buildNumber = readPropertyOrEnv("TRAVIS_BUILD_NUMBER", null);
-        }
-        if (buildNumber == null || buildNumber.equals("")) {
-            //try CIRCLE_BUILD_NUM
-            buildNumber = readPropertyOrEnv("CIRCLE_BUILD_NUM", null);
-        }
-
-        if (buildNumber != null && !(buildNumber.equals(""))) {
-            updates.put("build", buildNumber);
-        }
-
+        return new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss").format(new Date());
     }
-    private static String readPropertyOrEnv(String key, String defaultValue) {
+
+    private static String readPropertyOrEnv(String key) {
```

Comment:
What is the purpose of this function if it doesn't have default values?  Not that I think it should.